### PR TITLE
ISSUE-28: Add custom capability for Linagora/echo method

### PR DESCRIPTION
**Pull request by LanKhuat**

### CODE REVIEW

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by Arsnael** [comment]
Waiting for the James part to be finished before being able to merge this...

File: openpaas-james/jmap/src/main/scala/com/linagora/openpaas/james/jmap/CustomMethod.scala [hunk 1]
```diff
@@ -1,17 +1,34 @@
...
+}
+
+case class CustomCapabilityProperties() extends CapabilityProperties
+
+case class CustomCapability(properties: CustomCapabilityProperties = CustomCapabilityProperties(), identifier: CapabilityIdentifier = LINAGORA_ECHO) extends Capability
+
+class CustomCapabilitiesModule extends AbstractModule {
+
+  @ProvidesIntoSet
+  private def capability(): Capability = CustomCapability(new CustomCapabilityProperties, LINAGORA_ECHO)
```
Comment (chibenwa) [inline openpaas-james/jmap/src/main/scala/com/linagora/openpaas/james/jmap/CustomMethod.scala:26]: You do not need the new keyword here?

File: openpaas-james/jmap/src/main/scala/com/linagora/openpaas/james/jmap/CustomMethod.scala [hunk 2]
```diff
@@ -1,17 +1,34 @@
...
 import org.apache.james.jmap.method.{InvocationWithContext, Method}
 import org.apache.james.mailbox.MailboxSession
 import org.reactivestreams.Publisher
 import reactor.core.scala.publisher.SMono
 
+object CapabilityIdentifier {
+  val LINAGORA_ECHO: CapabilityIdentifier = "com:linagora:params:jmap:echo"
+}
+
+case class CustomCapabilityProperties() extends CapabilityProperties
```
Comment (chibenwa) [inline openpaas-james/jmap/src/main/scala/com/linagora/openpaas/james/jmap/CustomMethod.scala:19]: case object ?

File: openpaas-james/jmap/src/main/scala/com/linagora/openpaas/james/jmap/CustomMethod.scala [hunk 3]
```diff
@@ -1,17 +1,34 @@
...
+
+case class CustomCapability(properties: CustomCapabilityProperties = CustomCapabilityProperties(), identifier: CapabilityIdentifier = LINAGORA_ECHO) extends Capability
```
Comment (chibenwa) [inline openpaas-james/jmap/src/main/scala/com/linagora/openpaas/james/jmap/CustomMethod.scala:21]: Case object ?

**Pull request merged**